src/lib/utils/md.ts

```diff
@@ -60,9 +63,12 @@ const getPostSlugs = (dir: string, files: string[] = []) => {
         // If it is a .md file (allowed content page), push the path to the files array
         for (const page of temporalAllowedPages) {
           const fullPagePath = join(CURRENT_CONTENT_DIR, page)
+
           if (name.includes(fullPagePath)) {
             files.push(
-              name.replace(CURRENT_CONTENT_DIR, "").replace("/index.md", "")
+              fullPagePath
+                .replace(CURRENT_CONTENT_DIR, "")
+                .replace("/index.md", "")
```

Comment:
`fs.readdirSync(contentDir)` will give you a list of file paths relative to the dir you pass, not the absolute path.

```sh
# for example
> fs.readdirSync('./public/content/about')

# returns
[ 'index.md' ]
```

and then you are doing `const name = join(contentDir, file)` which will result in the path you need.

Reply:
I'm not sure I follow @pettinarip... using `name` (`join(contentDir, file)`) would end up with the full path vs just the slug, ie. `/repo/public/content/about/index.md` instead of just the `/about` slug. To my knowledge we only want the slug here... Am I missing something?

Reply:
Currently we need te full filepath to compute, recursively, all the paths combinations (eg: `/community/`, `/community/events`, etc). This is what the first loop is doing, when we find a .md file, we know it's a leaf node and can be used as a path. The second loop is checkin which paths could be added, as we're not building all pages but the ones from the`temporalAllowedPages` list only, for now. When we finish migrating all pages this list will be removed and this logic would need to be updated, as the second loop won't be required anymore.

Then, from each filepath, we're parsing it to get the clean slug we need for `GetStaticPaths`, this is what the `.replace` are for.

Reply:
You are right guys. For some reason, I thought that `name` was === to `fullPagePath.replace(CURRENT_CONTENT_DIR, "").replace("/index.md", "")`